types/mui-datatables/index.d.ts

```diff
@@ -159,120 +171,132 @@ export interface MUIDataTableIsRowCheck {
     ];
 }
 
-export interface MUIDataTableOptions {
-    caseSensitive?: boolean;
-    count?: number;
-    customFilterDialogFooter?: (filterList: any[]) => React.ReactNode;
-    customFooter?: (
+export type MUIDataTableOptions = Partial<{
+    caseSensitive: boolean;
+    confirmFilters: boolean;
+    count: number;
+    customFilterDialogFooter: (filterList: any[], applyNewFilters?: (...args: any[]) => any) => React.ReactNode;
+    customFooter: (
         rowCount: number,
         page: number,
         rowsPerPage: number,
         changeRowsPerPage: (page: string | number) => void,
         changePage: (newPage: number) => void
     ) => React.ReactNode;
-    customRowRender?: (data: any[], dataIndex: number, rowIndex: number) => React.ReactNode;
-    customSearch?: (searchQuery: string, currentRow: any[], columns: any[]) => boolean;
-    customSearchRender?: (
+    customRowRender: (data: any[], dataIndex: number, rowIndex: number) => React.ReactNode;
+    customSearch: (searchQuery: string, currentRow: any[], columns: any[]) => boolean;
+    customSearchRender: (
         searchText: string,
         handleSearch: any,
         hideSearch: any,
         options: any
     ) => React.Component | JSX.Element;
-    customSort?: (data: any[], colIndex: number, order: string) => any[];
-    customToolbar?: () => React.ReactNode;
-    customToolbarSelect?: (
-        selectedRows: {
-            data: Array<{ index: number; dataIndex: number }>;
-            lookup: { [key: number]: boolean };
-        },
+    customSort: (data: any[], colIndex: number, order: string) => any[];
+    customTableBodyFooterRender: (options: { data: any[]; selectableRows: SelectableRows; columns: any[] }) => any;
+    customToolbar: () => React.ReactNode;
+    customToolbarSelect: (
+        selectedRows: { data: Array<{ index: number; dataIndex: number }>; lookup: { [key: number]: boolean } },
         displayData: Array<{ data: any[]; dataIndex: number }>,
         setSelectedRows: (rows: number[]) => void
     ) => React.ReactNode;
-    disableToolbarSelect?: boolean;
-    download?: boolean;
-    downloadOptions?: Partial<{
+    download: boolean;
+    downloadOptions: Partial<{
         filename: string;
         separator: string;
         filterOptions: Partial<{ useDisplayedColumnsOnly: boolean; useDisplayedRowsOnly: boolean }>;
     }>;
-    elevation?: number;
-    expandableRows?: boolean;
-    expandableRowsOnClick?: boolean;
-    filter?: boolean;
-    filterType?: FilterType;
-    fixedHeader?: boolean;
-    fixedHeaderOptions?: {
```

Comment:
same as with sort order, should that one be only marked wiht `@deprecated` JSDoc?
https://github.com/gregnb/mui-datatables/blob/3.1.4/docs/v2_to_v3_guide.md#fixedheaderoptions-is-now-fixedheader-and-fixedselectcolumn
> The table will still accept the fixedHeaderOptions option, but it will print a deprecation notice when run in development mode.

Both ways of handling this are acceptable I think, but usually deprecation allows to migrate code in somehow slower fashion compared to tsc approach.